lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -178,6 +178,51 @@ class DecomposeAten_SoftmaxBackwardDataOp
 };
 } // namespace
 
+// Aten_LogSoftmaxBackwardDataOp(gradOutput, output, dim) =>
+//    result = gradOutput - (exp(output) * sum(gradOutput, dim))
+namespace {
+class DecomposeAten_LogSoftmaxBackwardDataOp
+    : public OpRewritePattern<Aten_LogSoftmaxBackwardDataOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten_LogSoftmaxBackwardDataOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    MLIRContext *context = op.getContext();
+    Value gradOutput = op.grad_output();
+    Value output = op.output();
+    Value dim = op.dim();
+
+    BaseTensorType tensorType = gradOutput.getType().cast<BaseTensorType>();
+    if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
+      return rewriter.notifyMatchFailure(op, "Only support floating type");
+
+    Value expOut =
+        rewriter.create<AtenExpOp>(loc, tensorType, output);
+    Value sum =
+        createAtenSum(rewriter, loc, op, gradOutput, dim, /*keepDim=*/true);
+    if (!sum)
+      return failure();
+
+    auto broadcastSizeType = Torch::ListType::get(Torch::IntType::get(context));
+    Value broadcastSize =
+        rewriter.create<AtenSizeOp>(loc, broadcastSizeType, output);
+    Value sumBroadcast =
+        rewriter.create<AtenBroadcastToOp>(loc, tensorType, sum, broadcastSize);
+    Value temp =
+        rewriter.create<AtenMulTensorOp>(loc, tensorType, expOut, sumBroadcast);
+
+    Value alpha =
```

Comment:
I think we now use AtenSubTensorOp often enough it probably worth to add a helper `createAtenTensorSub`  takes 2 tensors as arguments.

Reply:
Cool, I will add that.

Reply:
Done.

Reply:
This one can also use the helper :)

https://github.com/llvm/torch-mlir/blob/ea7a30f9b9b14f469024db35ce16bea2a5ee9548/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L214-L217